pkg/util/kube/predicates.go

```diff
@@ -108,25 +108,30 @@ func (r *vmForVMClassPredicate) Generic(e event.GenericEvent) bool {
 func (r *vmForVMClassPredicate) matches(obj runtime.Object) (bool, error) {
 	obj = obj.DeepCopyObject() // do not mutate the original
 
-	vm, err := r.getVM(obj)
+	object, kind, err := r.getWatchedObj(obj)
 	if err != nil {
 		return false, err
 	}
 
+	// if the object is not VirutalMachine, then pass to the xxxToVMMapperFn for further processing.
```

Comment:
Please ensure comments wrap at 80chars. Also, please capitalize `If the object` so sentences are grammatically correct.